src/picai_prep/dcm2mha.py

```diff
@@ -358,7 +408,13 @@ def process_and_write(self, output_dir: Path):
 
 
 class Dicom2MHAConverter:
-    def __init__(self, input_dir: PathLike, output_dir: PathLike, dcm2mha_settings: Union[PathLike, Dict]):
+    def __init__(
+        self,
+        input_dir: PathLike,
+        output_dir: PathLike,
+        dcm2mha_settings: Union[PathLike, Dict],
```

Comment:
Suggestion:

Make Dicom2MHASettings public, Dicom2MHAConverter then instead requires a Dicom2MHASettings object, rewrite Dicom2MHASettings to accept the json Path instead + the match functions and anything else we might think of later. 

also, I'm not sure how you can give `values_match_func`, I only see the `metadata_match_func` parameter.